app/Jobs/SendPostToCustomerIo.php

```diff
@@ -2,60 +2,67 @@
 
 namespace Rogue\Jobs;
 
-use DoSomething\Gateway\Blink;
 use Illuminate\Bus\Queueable;
 use Illuminate\Contracts\Queue\ShouldQueue;
 use Illuminate\Foundation\Bus\Dispatchable;
 use Illuminate\Queue\InteractsWithQueue;
 use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Redis;
 use Rogue\Models\Post;
+use Rogue\Services\CustomerIo;
 
 class SendPostToCustomerIo implements ShouldQueue
 {
     use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
 
     /**
-     * The post to send to Customer.io.
+     * Delete the job if its models no longer exist.
      *
-     * @var Post
+     * @var bool
      */
-    protected $post;
+    public $deleteWhenMissingModels = true;
 
     /**
-     * Whether or not to log that this Post was sent to Customer.io, via Blink.
+     * The post to send to Customer.io.
      *
-     * @var bool
+     * @var Post
      */
-    protected $log;
+    protected $post;
 
     /**
      * Create a new job instance.
      *
      * @return void
      */
-    public function __construct(Post $post, $log = true)
+    public function __construct(Post $post)
     {
         $this->post = $post;
-        $this->log = $log;
     }
 
     /**
      * Execute the job.
      *
      * @return void
      */
-    public function handle(Blink $blink)
+    public function handle(CustomerIo $customerIo)
     {
-        // Check if the post still exists before sending (might have been deleted immediately if created in Runscope test).
-        if ($this->post && $this->post->signup) {
```

Comment:
I was able to replace this conditional with Laravel's new [`$deleteWhenMissingModels`](https://laravel.com/docs/6.x/queues#ignoring-missing-models), above.